src/communication/metrics.hpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+#include <atomic>
+#include <map>
+#include <mutex>
+#include <shared_mutex>
+#include <string>
+#include <vector>
+
+#include <json/json.hpp>
+
+namespace memgraph::communication {
+
+class BoltMetrics {
+ public:
+  enum class ConnectionType { kAnonymous = 0, kBasic, Count };
+  static constexpr std::array<std::string_view, (int)ConnectionType::Count> ct_to_str = {"anonymous", "basic"};
+  static std::string ConnectionTypeStr(ConnectionType type) { return std::string(ct_to_str[(int)type]); }
```

Comment:
You could apply the same pattern as with IsolationMode and StorageMode strings but this can also pass

Reply:
The IsolationLevel and StorageMode are trying to parse strings and generate enums. Here I am trying to quickly get name of the enum. 
If there is something better or if I missed the exact code, could you paste the code you are thinking of?